pkg/printers/humanreadable.go

```diff
@@ -291,16 +293,43 @@ func printHeader(columnNames []string, w io.Writer) error {
 }
 
 // PrintObj prints the obj in a human-friendly format according to the type of the obj.
-// TODO: unify the behavior of PrintObj, which often expects single items and tracks
-// headers and filtering, with other printers, that expect list objects. The tracking
-// behavior should probably be a higher level wrapper (MultiObjectTablePrinter) that
-// calls into the PrintTable method and then displays consistent output.
 func (h *HumanReadablePrinter) PrintObj(obj runtime.Object, output io.Writer) error {
-	if w, found := output.(*tabwriter.Writer); !found && !h.skipTabWriter {
+	if meta.IsListType(obj) {
+		items, err := meta.ExtractList(obj)
+		if err != nil {
+			return err
+		}
+
+		if errs := runtime.DecodeList(items, h.decoder); len(errs) > 0 {
+			return utilerrors.NewAggregate(errs)
+		}
+
+		if len(h.options.SortBy) > 0 && len(items) > 1 {
+			// TODO: questionable
+			if _, err = SortObjects(h.decoder, items, h.options.SortBy); err != nil {
+				return err
+			}
+		}
+
+		for _, obj := range items {
+			if err := h.PrintObj(obj, output); err != nil {
+				return err
+			}
+		}
+		return nil
+	}
+
+	if h.options.WithKind {
+		gk := GetObjectGroupKind(obj, scheme.Scheme)
```

Comment:
Does not work yet. Currently prints `<unknown>/resource-name`. Any help here is appreciated